TFNetworkLayer.py

```diff
@@ -3345,12 +3345,19 @@ def __init__(self, axis, dim=1, **kwargs):
     data = self.input_data
     if isinstance(axis, int):
       data = data.copy_as_batch_major()
-    axis = self._get_axis(data=data, axis=axis)
+    axis_index = self._get_axis(data=data, axis=axis)
     from TFUtil import expand_dims_unbroadcast
-    self.output.placeholder = expand_dims_unbroadcast(data.placeholder, axis=axis, dim=dim)
+    self.output.placeholder = expand_dims_unbroadcast(data.placeholder, axis=axis_index, dim=dim)
     self.output.size_placeholder = {
-      (i if (data.get_batch_axis(i) < axis) else i + 1): v
+      (i if (data.get_batch_axis(i) < axis_index) else i + 1): v
       for (i, v) in data.size_placeholder.items()}
+    if axis in ["spatial", "time", "T"]:
+      self.output.time_dim_axis = axis_index
```

Comment:
It should not be overridden if it was already set before.

Reply:
RETURNN does not support multiple time dimensions, right? So we could assert that it is not yet set.

Reply:
Well, it supports, kind of. `time_dim_axis` can only be a single axis. But that does not matter very much in many cases. It can have multiple spatial axes, and it also can have multiple axes with variable length. And when you remove the axis which is currently `time_dim_axis` (e.g. by `ReduceLayer` or so), and there is another spatial axis, it usually will switch `time_dim_axis` over to the next one.